src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ReflectionXmlFormatWriter.cs

```diff
@@ -210,13 +211,25 @@ internal void ReflectionWriteCollection(XmlWriterDelegator xmlWriter, object obj
                     IEnumerator nonGenericDictEnumerator = (IEnumerator)new CollectionDataContract.DictionaryEnumerator(((IDictionary)obj).GetEnumerator());
                     enumerator = nonGenericDictEnumerator;
                 }
+                else if (collectionDataContract.Kind == CollectionKind.GenericCollection)
+                {
+                    elementType = collectionDataContract.ItemType;
+                }
                 else
```

Comment:
This 'else' has changed scope slightly with the insertion of the new code above.  Is it still correct for this to be the 'else' for the new 'if' above?